recipe/install-conda-specs.sh

```diff
@@ -0,0 +1,31 @@
+set -ex
+export CHOST="${gcc_machine}-${gcc_vendor}-linux-gnu"
+specdir=$PREFIX/lib/gcc/$CHOST/${gcc_version}
+if [[ "$cross_target_platform" == "$target_platform" ]]; then
+    install -Dm644 -T ${SRC_DIR}/build/gcc/specs $specdir/conda.specs
+
+    # Add specs when we're not cross compiling so that the toolchain works more like a system
+    # toolchain (i.e. conda installed libs can be #include <>'d and linked without adding any
+    # cmdline args or FLAGS and likewise the assumptions we have about rpath are built in)
+    #
+    # THIS IS INTENDED as a safety net for casual users who just want the native toolchain to work.
+    # It is not to be relied on by conda-forge package recipes and best practice is still to set the
+    # appropriate FLAGS vars (either via compiler activation scripts or explicitly in the recipe)
+    #
+    # We use double quotes here because we want $PREFIX and $CHOST to be expanded at build time
+    #   and recorded in the specs file.  It will undergo a prefix replacement when our compiler
+    #   package is installed.
+    sed -i -e "/\*link_command:/,+1 s+%.*+& %{\!static:-rpath ${PREFIX}/lib -rpath-link ${PREFIX}/lib -disable-new-dtags} -L ${PREFIX}/lib+" $specdir/conda.specs
```

Comment:
We currently have this in LDFLAGS that get installed by the binutils activation script https://github.com/conda-forge/ctng-compiler-activation-feedstock/blob/93ab56f85e1aac593fc3980fb33cacec25b01b4b/recipe/conda_build_config.yaml#L87-L96.  I'm surprised that we don't have it for the non-x86 linux platforms.

`-disable_new_dtags` (and the default behavior in newer binutils `-enable_new_dtags`) control whether `-rpath` sets  `DT_RPATH` entry in the ELF or `DT_RUNPATH`.  `DT_RPATH` and `DT_RUNPATH` are similar but slightly different.  `DT_RPATH` is highest priority path used by the dynamic loader and can not be overriden by `LD_LIBRARY_PATH` or the system config file for the dynamic linker.  This is explained in better detail in the [manpage for `ld.so(8)`](https://man7.org/linux/man-pages/man8/ld.so.8.html). 

There is also a significant difference in the behavior in that `DT_RPATH` applies to all subsequent dependencies that need to be looked up but `DT_RUNPATH` is only applied to direct dependencies and has to be set at each level of resolving.  (i.e. if you depend on a DSO, it has to also have DT_RUNPATH set).

When binutils defaulted to including the newer DT_RUNPATH in addition to the old DT_RPATH, it made everything I was building break because most of the libraries weren't also compiled with DT_RUNPATH and their dynamic dependencies wouldn't get resolved using DT_RPATH because it gets ignored when you have both DT_RPATH and DT_RUNPATH set.

I'm still scratching my head as to why the binutils-feedstock doesn't use `-disable-new-dtags` in the LDFLAGS it sets for the non-x86 linuxen.  Perhaps I shouldn't have added it into the default specfile.

Reply:
I'm still looking into why `linux-ppc64le`, `linux-aarch64` and `linux-s390x` don't use `-disable_new_dtags` but `linux-32` and `linux-64` do.  It's been that way since the beginning of the current conda-forge/ctng-compiler-activation-feedstock.  I'm not sure where the older history lives.

Reply:
Ray's old activation scripts in https://github.com/conda-archive/conda-recipes/tree/master/cross-compilers didn't have `-disable_new_dtags` in them for anything.  My personal fork of the old conda-recipes repo doesn't even have the `cross-compilers` recipes in it.

Reply:
No clue. We can add it to LDFLAGS

Reply:
Captured this in an issue on the activation feedstock.  https://github.com/conda-forge/ctng-compiler-activation-feedstock/issues/72